javascript_nodejs/14.nlp-with-dispatch/dialogs/weather/index.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const { LuisRecognizer } = require('botbuilder-ai');
+
+// LUIS intent names. you can get this from the .lu file.
+const GET_CONDITION_INTENT = 'Get_Weather_Condition';
+const GET_FORECAST_INTENT = 'Get_Weather_Forecast';
+const NONE_INTENT = 'None';
+
+// LUIS entity names. 
+const LOCATION_ENTITY = 'Location';
+const LOCATION_PATTERNANY_ENTITY = 'Location_PatternAny';
+
+// this is the LUIS service type entry in the .bot file.
+const WEATHER_LUIS_CONFIGURATION = 'weather.luis';
+
+class Weather {
+    /**
+     * 
+     * @param {Object} botConfig bot configuration from .bot file
+     */
+    constructor(botConfig) {
+        if(!botConfig) throw ('Need bot config');
+        
+        // add recogizers
+        const luisConfig = botConfig.findServiceByNameOrId(WEATHER_LUIS_CONFIGURATION);
+        if(!luisConfig || !luisConfig.appId) throw (`Weather LUIS model not found in .bot file. Please ensure you have all required LUIS models created and available in the .bot file. See readme.md for additional information\n`);
+        this.luisRecognizer = new LuisRecognizer({
+            applicationId: luisConfig.appId,
+            azureRegion: luisConfig.region,
+            // CAUTION: Its better to assign and use a subscription key instead of authoring key here.
+            endpointKey: luisConfig.authoringKey
+        });
+    }
+    /**
+     * 
+     * @param {Object} context context object
+     */
+    async onTurn(context) {
+        // make call to LUIS recognizer to get home automation intent + entities
+        const weatherResults = await this.luisRecognizer.recognize(context);
+        const topWeatherIntent = LuisRecognizer.topIntent(weatherResults);
+        // get location entity if available.
```

Comment:
ends in a ".". Means it should start with a "G". :)

Reply:
Fixed.